tree/tree/test/TChainParsing.cxx

```diff
@@ -184,6 +185,92 @@ TEST(TChainParsing, GlobbingWithNonExistingDir)
    }
 }
 
+std::vector<std::string> GetFileNamesVec(const TObjArray *chainFiles)
+{
+   std::vector<std::string> chainFileNames;
+   chainFileNames.reserve(chainFiles->GetEntries());
+   for (const auto *obj : *chainFiles) {
+      chainFileNames.push_back(obj->GetTitle());
+   }
+   return chainFileNames;
+}
+
+TEST(TChainParsing, RecursiveGlob)
+{
+   // Need to change working directory to ensure "*" only adds files we create here.
+   auto testDir = "testdir";
+   ASSERT_EQ(gSystem->mkdir(testDir), 0);
+   gSystem->ChangeDirectory(testDir);
+
+   const auto *cwd = gSystem->WorkingDirectory();
+
+   ASSERT_EQ(gSystem->mkdir("testglob"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir1"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir1/subdir11"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir2"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir2/subdir21"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir3"), 0);
+
+   // Unsorted to also test sorting the final list of globbed files.
+   std::vector<std::string> fileNames{"0a.root",
+                                      "testglob/subdir1/1a.root",
+                                      "testglob/subdir1/1b.root",
+                                      "testglob/subdir3/3a.root",
+                                      "testglob/subdir2/subdir21/21a.root",
+                                      "testglob/subdir2/subdir21/21b.root",
+                                      "testglob/subdir1/subdir11/11b.root",
+                                      "testglob/subdir1/subdir11/11a.root"};
+   for (const auto &fileName : fileNames) {
+      FillTree(fileName.c_str(), "events");
+   }
+
+   TChain nodir, none, nested, globDir;
```

Comment:
Many people, like me or [clang-tidy](https://clang.llvm.org/extra/clang-tidy/checks/readability/isolate-declaration.html), don't like to have multiple declarations per line because it makes the code harder to read.